src/odemis/acq/stream/_base.py

```diff
@@ -561,6 +608,76 @@ def _getDetectorVA(self, vaname):
                 raise AttributeError(u"Detector has not VA %s" % (vaname,))
             return hwva
 
+    def _linkHwAxes(self):
+        """"
+        Link the axes to the hardware components
+        If local axis Vigilant Attributes's are specified, write the values of the local axis VA
+        to the real hardware
+        """
+
+        if hasattr(self, "axis_map") and self.axis_map:
+
+            logging.info("Moving linked HW axes")
+
+            moving_axes = []
+            for va_name, (axis_name, actuator) in self.axis_map.items():
+                try:
+                    local_pos_va = self._axis_vas[va_name]
+                    pos = local_pos_va.value
+                    logging.info("Moving actuator %s axis %s to position %s.", actuator.name, axis_name, pos)
+                    f = actuator.moveAbs({axis_name: pos})
+                    f.add_done_callback(self._onAxisMoveDone)
+                    moving_axes.append(f)
+                    # subscribe to update the axis when the stream plays
+                    updater = functools.partial(self._update_linked_axis, axis_name)
+                    self._axisvaupdaters[va_name] = updater
+                    local_pos_va.subscribe(updater)
+                except Exception:
+                    logging.exception("Failed to move actuator %s axis %s.", actuator.name, axis_name)
+
+            for f in moving_axes:
+                try:
+                    f.result()
+                except Exception:
+                    logging.exception("Failed to move axis.")
+                    
+    def _onAxisMoveDone(self, f):
+        """
+         Callback method, which checks that the move is actually finished.
+        :param f: (future)
+        """
+        try:
+            f.result()
+        except Exception:
+            logging.exception("Failed to move axis.")
+
+    def _update_linked_axis(self, ax_name, pos):
+        """ Update the value of a linked hardware axis VA
+            when the stream is active
+        """
+        if not self.is_active.value:
+            return
+        try:
+            act = self.axis_map[ax_name][1]
+            logging.info("Moving actuator %s to position %s.", act.name, pos)
+            f = act.moveAbs({ax_name: pos})
+            # TODO: Move this to a separate thread to make sure it doens't block the GUI
+            # it would be good to have the caller know when the axis is done moving.
```

Comment:
To clarify, I'd suggest to write something more like:
```
TODO: ideally, it would block, so that the the caller knows when the move is complete.
   However, this requires that the GUI calls this function is a separate thread.
```